src/charm.py

```diff
@@ -740,6 +721,144 @@ def _gen_wp_config(self):
         )
         return "\n".join(wp_config)
 
+    def _container(self):
+        return self.unit.get_container("wordpress")
+
+    def _wordpress_service_exists(self):
+        return "wordpress" in self._container().get_plan().services
+
+    def _stop_server(self):
+        """Stop WordPress (apache) server
+
+        This operation is idempotence.
+        """
+        if self._wordpress_service_exists():
+            self._container().stop("wordpress")
+
+    def _wp_is_installed(self):
+        """Check if WordPress is installed (check if WordPress related tables exist in database)"""
+        process = self._container().exec(["wp", "core", "is-installed"])
+        try:
+            process.wait()
+            return True
+        except ops.pebble.ExecError:
+            return False
+
+    def _wp_install_cmd(self):
+        """Generate wp-cli command used to install WordPress on database"""
+        initial_settings = yaml.safe_load(self.model.config["initial_settings"])
+        admin_user = initial_settings.get("user_name", "admin_username")
```

Comment:
It looks like the default settings define a value for `user_name` and `admin_email` in the `config.yaml` file, should the same values be used here? Potentially even just read the values from that file so that they are only defined in one place?

Reply:
The default value for `initial_settings` is `"user_name: admin\nadmin_email: devnull@example.com"`. If `initial_setting` being set to `""`, it will be equivalent to `"user_name: admin_username\nadmin_email: name@example.com"`. It's confusing here, but for the sake of backward compatibility, just implement as is.

Reply:
Does the charm currently use `admin_username` and `name@example.com`?

Reply:
if user set the `initial_setting` config to `""` explicitly, then it will use `admin_username` and `name@example.com`

Reply:
Ok